ginga/misc/plugins/MultiDim.py

```diff
@@ -493,6 +532,46 @@ def redo(self):
 
         self.w.hdu.set_enabled(len(self.fits_f) > 0)
 
+    def save_slice_cb(self):
+        target = SaveDialog(title='Save slice', extfilter='PNG image (*.png)').get_path()
+        if target:
+            hival = self.fitsimage.get_cut_levels()[1]
+            plt.imsave(target, self.curr_slice, vmax=hival, cmap=plt.get_cmap('gray'), origin='lower')
+            self.fv.showStatus("Successfully saved slice")
+
+    def save_movie_cb(self):
+        start = int(self.w.start.get_text())
+        end = int(self.w.end.get_text())
+        if not start or not end:
+            return
+        elif start < 0 or end > self.play_max:
+            self.fv.showStatus("Wrong slice index")
+            return
+        elif start > end:
+            self.fv.showStatus("Wrong slice order")
+            return
+
+        if start == 1:
+            start = 0
+
+        target = SaveDialog('Save Movie', 'AVI movie (*.avi)').get_path()
+        if target:
+            self.save_movie(start, end, target)
+
+    def save_movie(self, start, end, target):
+        hdu = self.fits_f[self.curhdu]
+        loval, hival = self.fitsimage.get_cut_levels()
+        data = np.array(hdu.data, copy=True).clip(loval, hival)
+
+        data_rescaled = ((data - loval) * 255 / (hival - loval)).astype(np.uint8, copy=False)
+        H, W = data.shape[1:]
+
+        video = VideoSink((H, W), target)
+        for i in xrange(start, end):
+            video.run(np.flipud(data_rescaled[i]))
+        video.close()
+
```

Comment:
Wondering if we can make the `VideoSink` class act like a context manager, so you could do:
```python
with VideoSink((H, W), target) as video:
    for i in xrange(start, end):
        video.run(np.flipud(data_rescaled[i]))
```
The nice thing about context managers is that they automatically clean up and don't leave open files or processes around if the encoding craps out in the middle due to some error...